### ISSUE TITLE
bug: gpg-agent using removed "gtk2" pinentry flavor as default

### ISSUE THREAD

**Issue by TomaSajt** [nix-community/home-manager#4804]
### Are you following the right branch?

- [X] My Nixpkgs and Home Manager versions are in sync

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Issue description

The `gtk2` flavor has been removed from Nixpkgs in [this commit](https://github.com/NixOS/nixpkgs/commit/edb8b7994c383d3754cbb03b2bdb6016026e19cc) but Home Manager still uses it as a default

I only encountered an error with `gpg-agent`, but there might be some other modules, which I don't use, that have similar problems.

### Maintainer CC

@rycee

### System information

```markdown
- system: `"x86_64-linux"`
- host os: `Linux 6.1.69, NixOS, 24.05 (Uakari), 24.05.20231221.d6863cb`
- multi-user?: `yes`
- sandbox: `yes`
- version: `nix-env (Nix) 2.18.1`
- channels(root): `""`
- nixpkgs: `/nix/store/61gw1axgxrvx0bnxv2dla6zhkgwv45zr-source`
```

**Comment by pinpox** [comment]
I'm running into the same issue. What is a sensible default for non-gnome setups? Specifically I'm using sway and am not sure what to set it to

**Comment by inclyc** [comment]
This breaks my CI recently.

**Comment by rycee** [comment]
I switched to the `gnome3` pinentry years ago. It has worked without any issues and I don't use Gnome very much otherwise. The main thing to consider is that the system configuration needs

    services.dbus.packages = [ pkgs.gcr ];

I actually don't know if that is a hard requirement, maybe it is possible to get things to work with all user-side configuration. Might be safer to do on system-side, though, since more hardening options are available there.

**Comment by rycee** [comment]
Perhaps we should re-instate gtk2 as the default but emit a warning if used. Then people will have to set the option explicitly (to a non-gtk2 value) and we can in the future remove the default entirely without causing too much complaints. @SuperSandro2000, does that seem reasonable?

**Comment by pinpox** [comment]
I just changed it to `qt` and it works fine, but is quite ugly if you don't use qt otherwise. A gtk alternative without gnome dependencies would be nice. Maybe we could also just check if gnome is set and add the dependencies automatically to the config if so?

**Comment by bbenne10** [comment]
458544594ba7f0333cf5718045ee7a8eaf5de433 addresses this, but does so only for master/nixpkgs unstable. Is it possible to get this fix backported to release-23.11? 

Context: I use 23.11 and am trying to pick a "non-standard" pinentry program (pinentry-bemenu), which I should be able to do (with this fix) with:

```
services.gpg-agent.extraConfig = ''
  pinentry-program ${pkgs.pinentry-bemenu}/bin/pinentry-bemenu
'';
```

But without this change, this ends up providing *two* pinentry fields: the gtk2 one and pinentry-bemenu. BUT gpg-agent picks the first entry. Since `extraConfig` is appeneded, I simply can't fix this without providing the entire file by hand (and then I lose out on `enableZshIntegration` and the like...)

**Comment by TomaSajt** [comment]
This issue has been mostly solved by https://github.com/nix-community/home-manager/pull/4805,
and then recently, the module was reworked in https://github.com/nix-community/home-manager/pull/4895 so that there is no longer a string option to set.
I think I can safely close this issue.
